### ISSUE TITLE
Jupyter notebook magic not working (unicode error)

### ISSUE THREAD

**Issue by Didou09** [serge-sans-paille/pythran#693]
Hi,

Using the load_ext in te jupyter notebook to use pythran on a cell I got the following error (on Romeo during the advanced Python training in Jussieu):

```python
import pythran
%load_ext pythran.magic
```
Works fine, then:

```python
%%pythran
#pythran export add(int,int)
def add(i,j):
    return i+j
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-24-bd673c43d8bb> in <module>()
----> 1 get_ipython().run_cell_magic('pythran', '', '#pythran export add(int, int)\ndef add(i, j):\n    return i+j')

/apps/python/3/lib/python3.5/site-packages/IPython/core/interactiveshell.py in run_cell_magic(self, magic_name, line, cell)
   2101             magic_arg_s = self.var_expand(line, stack_depth)
   2102             with self.builtin_trap:
-> 2103                 result = fn(magic_arg_s, cell)
   2104             return result
   2105 

<decorator-gen-361> in pythran(self, line, cell)

/apps/python/3/lib/python3.5/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

/apps/python/3/lib/python3.5/site-packages/pythran/magic.py in pythran(self, line, cell)
     67                 '-fopenmp')
     68         m = hashlib.md5()
---> 69         m.update(cell)
     70         module_name = "pythranized_" + m.hexdigest()
     71         module_path = pythran.compile_pythrancode(module_name, cell, **kwargs)

TypeError: Unicode-objects must be encoded before hashing

```

**Comment by flothesof** [comment]
Hi, 
attempt at fixing this with pull request #692.
Best,
Florian

**Comment by serge-sans-paille** [comment]
Closed by #692
